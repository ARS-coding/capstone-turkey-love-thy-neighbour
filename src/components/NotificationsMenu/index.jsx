import React from "react";

import { Dropdown } from "react-bootstrap";

import { useSelector } from "react-redux";

import NotificationMenuItem from "../NotificationMenuItem";

import { ReactComponent as NotificationBellSVG } from "../../images/notification-bell.svg";

import { firestore } from "../../firebaseConfig";

import "./index.scss";

function NotificationMenu() {
  const invitationNotifications = useSelector(
    (state) => state.user.firestoreDoc?.invitationNotifications
  );
  const uid = useSelector((state) => state.user.authCred?.uid);

  function handleClearAll() {
    firestore
      .collection("users")
      .doc(uid)
      .update({ invitationNotifications: [] });
  }

  return (
    <Dropdown id="notifications-dropdown" className="order-lg-last">
      <Dropdown.Toggle variant="success" id="dropdown-basic">
        <NotificationBellSVG />
      </Dropdown.Toggle>

      <Dropdown.Menu>
        <Dropdown.Header className="d-flex justify-content-between align-items-center">
          Notifications
          <button
            type="button"
            className="clear-all-btn"
            onClick={handleClearAll}
          >
            Clear All
          </button>
        </Dropdown.Header>

        <Dropdown.Divider />

        {invitationNotifications?.length !== 0 &&
        invitationNotifications !== null &&
        invitationNotifications !== undefined ? (
          invitationNotifications.map((invitationNotificationObject) => (
            <NotificationMenuItem
              key={invitationNotificationObject.id}
              invitationNotificationObject={invitationNotificationObject}
            />
          ))
        ) : (
          <p className="ms-3">No current notifications...</p>
        )}
      </Dropdown.Menu>
    </Dropdown>
  );
}

export default NotificationMenu;
