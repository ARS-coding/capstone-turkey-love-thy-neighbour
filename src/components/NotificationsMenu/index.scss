@import "../../style/variables";

div#notifications-dropdown {
  position: relative;

  button.dropdown-toggle {
    background-color: $dark-blue;
    border: none;
    box-shadow: none !important;

    &::after {
      content: none;
    }

    &:hover {
      color: rgba(255, 255, 255, 0.75);

      svg {
        fill: rgba(255, 255, 255, 0.75);
      }
    }

    svg {
      fill: rgba(255, 255, 255, 0.55);
      transition: all ease-in-out 150ms;
    }
  }

  div.dropdown-menu {
    padding-bottom: 0;
    max-height: calc(100vh - 200px);
    min-height: 400px;
    min-width: 300px;
    overflow: scroll;

    position: absolute;
    top: 38px;

    @media screen and (max-width: 991px) {
      right: -125px;
    }
    @media screen and (min-width: 992px) {
      right: 0;
    }

    & > .dropdown-header {
      p {
        margin: 0 !important;
        color: black !important;
      }

      button.clear-all-btn {
        color: #6c757d;
        background-color: transparent;
        border: none;
        font-size: 10.5px;
      }
    }
    p {
      color: #6c757d;
    }
  }
}
