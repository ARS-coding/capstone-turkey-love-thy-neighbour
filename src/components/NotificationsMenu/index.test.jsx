import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import NotificationsMenu from ".";

describe("NotificationsMenu", () => {
  const data = {
    invitationNotificationObject: {
      message: "test",
      id: "test",
    },
  };
  const mockState = {
    user: {
      authCred: {
        uid: "Test Uid",
      },
      firestoreDoc: {
        invitationNotifications: [
          {
            id: "test",
            message: "test",
          },
        ],
      },
    },
  };
  const mockStore = configureStore();
  let store;
  let tree;

  beforeEach(() => {
    store = mockStore(mockState);
    tree = renderer
      .create(
        <Provider store={store}>
          <NotificationsMenu />
        </Provider>
      )
      .toJSON();
  });

  it("renders NotificationsMenu correctly", () => {
    expect(tree).toMatchSnapshot();
  });
});
