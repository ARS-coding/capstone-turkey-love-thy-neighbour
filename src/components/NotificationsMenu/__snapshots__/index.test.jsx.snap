// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsMenu renders NotificationsMenu correctly 1`] = `
<div
  className="dropdown"
  id="notifications-dropdown"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-success"
    disabled={false}
    id="dropdown-basic"
    onClick={[Function]}
    type="button"
  >
    <svg>
      notification-bell.svg
    </svg>
  </button>
</div>
`;
